### PR TITLE
Option to use number for user and/or group names

### PR DESCRIPTION
**Pull request by breezerider**
* add NumericUid and NumericGid to Tar struct
* writeFileToArchive: set Uname and Gname to empty string if NumericUid and NumericGid are true, respectively.

### CODE REVIEW

**Comment by breezerider** [comment]
No real use case on my side, just left it like that for flexibility. Primary use case is to prevent aliasing and this means both flags are set simultaneously. Do you think it makes sense to join them?

**Comment by mholt** [comment]
I think we should probably join them unless there's a valid use case for having them separated. We're still in alpha so we could change this later if there's a valid use case brought forth.

File: tar.go [hunk 1]
```diff
@@ -17,6 +17,9 @@ func init() {
 }
 
 type Tar struct {
+	// If true, leave only numeric user and group id
+	NumericUidGid bool
```
Comment (mholt) [inline tar.go:21]:
```suggestion
	// If true, preserve only numeric user and group id
	NumericUIDGID bool
```

(Just a nit, don't mind me. I'll still merge this in a moment.)

File: tar.go [hunk 2]
```diff
@@ -81,6 +84,10 @@ func (Tar) writeFileToArchive(ctx context.Context, tw *tar.Writer, file File) er
 		return fmt.Errorf("file %s: creating header: %w", file.NameInArchive, err)
 	}
 	hdr.Name = file.NameInArchive // complete path, since FileInfoHeader() only has base name
+	if t.NumericUidGid {
```
Comment (mholt) [inline tar.go:87]:
```suggestion
	if t.NumericUIDGID {
```

**Review by mholt — APPROVED** [review]
Thanks! This is good. I'm just changing `UidGid` to `UIDGID`, as dumb as that looks, because the convention I'm used to capitalizes initialisms :upside_down_face:

I appreciate the change!!

**Pull request merged**